File: app/src/main/java/com/shahin/movieapp/view/SplashActivity.java
package com.shahin.movieapp.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.shahin.movieapp.R;
import com.shahin.movieapp.viewModel.MainViewModel;

public class SplashActivity extends AppCompatActivity {

    private MainViewModel mMainViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        mMainViewModel = new ViewModelProvider(this).get(MainViewModel.class);
        mMainViewModel.getMoviesList();

        new Handler().postDelayed(() -> {
            Intent intent = new Intent(SplashActivity.this, MainActivity.class);
            startActivity(intent);
            finish();
        }, 5000);


    }
}

File: app/src/main/java/com/shahin/movieapp/ui/splash/SplashActivity.kt
package com.shahin.movieapp.ui.splash

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import androidx.lifecycle.ViewModelProvider
import com.shahin.movieapp.R
import com.shahin.movieapp.ui.movieList.MainActivity
import com.shahin.movieapp.viewModel.MainViewModel

class SplashActivity : AppCompatActivity() {
    private var mMainViewModel: MainViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_splash)
        mMainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        mMainViewModel!!.getMoviesList()
        Handler().postDelayed({
            val intent = Intent(this@SplashActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }, 5000)

    }

}